src/content/en/fundamentals/security/credential-management/retrieve-credentials.md

```diff
@@ -4,369 +4,396 @@ book_path: /web/fundamentals/_book.yaml
-
-    if (cred) {
-      if (cred.type == 'password') {
-        // Use `email` instead of `id` for the id
-        cred.idName = 'email';
-
-        // Append CSRF Token
-        var csrf_token = document.querySelector('#csrf_token').value;
-        var form = new FormData();
-        form.append('csrf_token', csrf_token);
-
-        // Append additional credential data to `.additionalData`
-        cred.additionalData = form;
-
-        // `POST` the credential object.
-        // id, password and the additional data will be encoded and
-        // sent to the url as the HTTP body.
-        fetch(url, {           // Make sure the URL is HTTPS
-          method: 'POST',      // Use POST
-          credentials: cred    // Add the password credential object
-        }).then(function() {
-          // continuation
-        });
-      }
-    }
+To implement federated login:
 
-You can do a similar thing by assigning a `URLSearchParams` object instead of
-a `FormData` to `.additionalData`. In this case, the whole credential object 
-is encoded using `application/x-www-form-urlencoded`.
+1. Authenticate the user with a third-party identity.
+2. Store the identity information.
+3. [Update UI or proceed to a personalized page](#update_ui) (same as auto sign-in).
 
-### Authenticate with an identity provider
+### Authenticate user with third-party identity
 
-To authenticate the user with an identity provider, simply use the 
-specific authentication flow with the `FederatedCredential`.
+When a user taps on a federated login button,
+run the specific identity provider authentication flow with the `FederatedCredential`.
 
 For example, if the provider is Google, use the
-[Google Sign-In JavaScript library](/identity/sign-in/web/):
+[Google Sign-In JavaScript library](https://developers.google.com/identity/sign-in/web/):
 
     // Instantiate an auth object
     var auth2 = gapi.auth2.getAuthInstance();
 
     // Is this user already signed in?
     if (auth2.isSignedIn.get()) {
       var googleUser = auth2.currentUser.get();
-      
+
       // Same user as in the credential object?
       if (googleUser.getBasicProfile().getEmail() === id) {
         // Continue with the signed-in user.
         return Promise.resolve(googleUser);
       }
     }
-    
+
     // Otherwise, run a new authentication flow.
     return auth2.signIn({
       login_hint: id || ''
     });
 
+Google Sign-In results in an ID token as a proof of authentication,
+which gets sent to the server to create a session.
 
-Google Sign-In results in an id token as a proof of authentication which
-you send to the server to create a session.
+### Store identity information
 
-For additional identity providers, please refer to respective documentation:
+Once authentication is done, you can store the identity information. The information you’ll store here is the id from the identity provider and a provider string that represents the identity provider (`name` and `iconURL` are optional).
```

Comment:
So I made a decision to not link to the first page, as it relates to Android Identity-- and I can't see ID specifically. Maybe it is there,  but as I couldn't make sense of it, I opted to go with the a link to the specification which is clear and matches the text.